etl/steps/data/garden/health/2023-04-18/wgm_mental_health.py

```diff
@@ -0,0 +1,339 @@
+
+
+def make_table(df: pd.DataFrame) -> Table:
+    """Create a new table from a dataframe."""
+    df = clean_df(df)
+    df = make_df_with_share_answers(df)
+    df = map_ids_to_labels(df)
+    df = filter_rows_with_low_participation(df)
+    df = final_formatting(df)
+    tb = Table(df, short_name=paths.short_name, underscore=True)
+    return tb
+
+
+def clean_df(df: pd.DataFrame) -> pd.DataFrame:
+    """Keep relevant columns, unpivot dataframe, harmonise country names."""
+    log.info("wgm_mental_health: cleaning dataframe")
+    # Relevant columns
+    columns_rel = list(MAPPING_COLUMN_NAMES.keys()) + list(MAPPING_QUESTION_VALUES.keys())
+    df = df[columns_rel]
+    # Rename columns
+    df = df.rename(columns=MAPPING_COLUMN_NAMES)
+    # Unpivot
+    log.info("wgm_mental_health: unpivotting dataframe")
+    df = df.melt(
+        id_vars=list(MAPPING_COLUMN_NAMES.values()),
+        value_vars=list(MAPPING_QUESTION_VALUES.keys()),
+        var_name="question",
+        value_name="answer",
+    )
+    # Set dtypes of all question columns to strings
+    df = df.astype({"question": str, "answer": str})
+    # Harmonise country names
+    log.info("wgm_mental_health: harmonize_countries")
+    df = geo.harmonize_countries(df=df, countries_file=paths.country_mapping_path)
+    return df
+
+
+def make_df_with_share_answers(df: pd.DataFrame) -> pd.DataFrame:
+    """Build dataframe with shares of answers to each questions.
+
+    Obtains values for all countries, continents and income groups. Also for various demographic groups (age, gender)
+    """
+    # Obtain dataframe for countries
+    log.info("wgm_mental_health: building dataframe for countries")
+    df_countries = _make_df_with_share_answers(df)
+    # Obtain dataframe for World
+    log.info("wgm_mental_health: building dataframe for World")
+    df_world = df.assign(country="World")
+    df_world = _make_df_with_share_answers(df_world, "weight_inter_country")
+    # Obtain dataframe for continents
+    log.info("wgm_mental_health: building dataframe for continents")
+    df_continents = _build_df_with_continents(df)
+    df_continents = _make_df_with_share_answers(df_continents, "weight_inter_country")
+    # Obtain dataframe for income groups
+    log.info("wgm_mental_health: building dataframe for income groups")
+    df_incomes = _build_df_with_incomes(df)
+    df_incomes = _make_df_with_share_answers(df_incomes, "weight_inter_country")
+    # Merge
+    df = pd.concat([df_countries, df_world, df_continents, df_incomes], ignore_index=True)
+    return df
+
+
+def _build_df_with_continents(df: pd.DataFrame) -> pd.DataFrame:
+    # Obtain dataframe for continents
+    df_continents = df.copy()
+    continents = ["Africa", "Asia", "Europe", "North America", "Oceania", "South America"]
+    df_continents["country"] = df_continents["country"].cat.add_categories(continents)
+    for continent in continents:
+        countries = geo.list_countries_in_region(continent)
```

Comment:
If you use this function, you need to include `wb_income` as a dependency in the dag (which is unfortunate, we have to work on these `geo` functions once and for all).